### ISSUE TITLE
The knowledge base needs to be able to add custom fields and use them for filtering in workflow knowledge retrieval, which would be a particularly useful feature

### ISSUE THREAD

**Issue by miludedeng** [langgenius/dify#9630]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

When I call a workflow, I need to filter specific data in the knowledge base in different scenarios. However, currently dify can only retrieve information from the entire knowledge base. I have to create multiple knowledge bases and workflows. But this will be a huge workload.

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by Yawen-1010** [comment]
Thank you for your valuable request. Adding Metadata to the knowledge base data may solve your problem. We are investigating internally. This part of the functionality is on the knowledge base product planning roadmap.

**Comment by liuzixiao666** [comment]
我也需要这个功能，非常需要。请问大概什么时候能上线该功能，根据不同的用户，我应当检索不同的数据集（或者说我可以根据我附带的变量选择某个数据集筛选出满足某些字段的数据）

**Comment by 143230** [comment]
yes, i need this fea too, is there anyway i can see your planning roadmap for this feature?

**Comment by nhha1602** [comment]
Any update this task ?

**Comment by christophersandmann** [comment]
This is still relevant.

**Comment by crazywoola** [comment]
See https://docs.dify.ai/guides/knowledge-base/metadata.
